tests/41end-to-end-keys/01-upload-key.pl

```diff
@@ -162,28 +144,48 @@
       })
    };
 
-push our @EXPORT, qw( matrix_put_e2e_keys );
+push our @EXPORT, qw( matrix_put_e2e_keys matrix_get_e2e_keys );
 
 sub matrix_put_e2e_keys
 {
    # TODO(paul): I don't really know what's parametric about this
    my ( $user, %params ) = @_;
 
    my $device_keys = $params{device_keys} // {};
+   $device_keys->{user_id} = $user->user_id;
+   $device_keys->{device_id} = $user->device_id;
 
    do_request_json_for( $user,
       method => "POST",
       uri    => "/unstable/keys/upload",
 
       content => {
-         device_keys => {
-            user_id => $user->user_id,
-            device_id => $user->device_id,
-            device_keys => $device_keys,
-         },
+         device_keys => $device_keys,
```

Comment:
looks like you're changing the API for this function, but haven't changed all the placed it is used?

Reply:
AFAICT, other than the new uses in the cross-signing test, the only time the `device_keys` is specified is adding a dummy value to make synapse think that there's a device update.  This is based on the fact that the spec doesn't define a nested `device_keys` property, nor an `updated` property within `device_keys`.  So I don't think that the changed API makes a difference.  (I re-ran the tests, and they seem to pass.)

Reply:
fair enough!